lib/timescale/migration.ex

```diff
@@ -7,46 +7,185 @@ defmodule Timescale.Migration do
+    If this is set, the `to_version` option will be ignored.
+    If you need to set the schema and upate to a version, call this function twice with each option.
+  - `:to_version` `string` In the up direction, the version of the extension to update to.
+  """
+  defmacro update_timescaledb_extension(opts \\ []) do
+    if schema = opts[:schema] do
+      quote bind_quoted: [schema: schema] do
+        Ecto.Migration.execute("ALTER EXTENSION timescaledb SET SCHEMA #{schema}")
+      end
+    else
+      to_version = if opts[:to_version], do: "TO '#{opts[:to_version]}'"
+
+      quote bind_quoted: [to_version: to_version] do
+        Ecto.Migration.execute(trim("ALTER EXTENSION timescaledb UPDATE #{to_version}"), "")
+      end
     end
   end
 
   @doc """
   Updates the existing [TimescaleDB toolkit](https://docs.timescale.com/timescaledb/latest/how-to-guides/hyperfunctions/install-toolkit/#install-and-update-timescaledb-toolkit) extension
+
+  Irreversible migration.
+
+  Options:
+  - `:to_version` `string` In the up direction, the version of the extension to update to.
   """
-  defmacro update_timescaledb_toolkit_extension do
-    quote do
-      Ecto.Migration.execute("ALTER EXTENSION timescaledb_toolkit UPDATE")
+  defmacro update_timescaledb_toolkit_extension(opts \\ []) do
+    if opts[:schema],
+      do:
+        raise(Timescale.MigrationArgError,
+          function: :update_timescaledb_toolkit_extension,
+          invalid_args: [:schema]
+        )
+
+    to_version = if opts[:to_version], do: "TO '#{opts[:to_version]}'"
+
+    quote bind_quoted: [to_version: to_version] do
+      Ecto.Migration.execute(trim("ALTER EXTENSION timescaledb_toolkit UPDATE #{to_version}"))
     end
   end
 
   @doc """
   Drops the [TimescaleDB toolkit](https://docs.timescale.com/timescaledb/latest/how-to-guides/hyperfunctions/install-toolkit/#install-and-update-timescaledb-toolkit) as a Postgres Extension
+
+  Options:
+  - `:cascade` `true|false` Forcibly remove dependant objects.
+  - `:if_exists` `true|false` Drop extension only if it exists. Default: `true`
+
+  Down direction options:
+  - `:if_not_exist` `true|false` Create extension only if it doesn't exists. Default: `true`
+  - `:schema` `string` The name of the schema in which to install the extensions' objects.
+  - `:version` `string` The version of the extension to install.
   """
-  defmacro drop_timescaledb_toolkit_extension do
-    quote do
-      Ecto.Migration.execute("DROP EXTENSION IF EXISTS timescaledb_toolkit CASCADE")
+  defmacro drop_timescaledb_toolkit_extension(opts \\ []) do
+    if_exists = if Keyword.get(opts, :if_exists, true), do: "IF EXISTS"
+    if_not_exists = if Keyword.get(opts, :if_not_exists, true), do: "IF NOT EXISTS"
+    cascade = if opts[:cascade], do: "CASCADE"
+    schema = if opts[:schema], do: "SCHEMA #{opts[:schema]}"
+    version = if opts[:version], do: "VERSION '#{opts[:version]}'"
```

Comment:
I suppose no one is probably generate migrations off of user input, but we still may want to escape these values